### ISSUE TITLE
Error: Topics missing from index: municipality_key_2022

### ISSUE THREAD

**Issue by pitkant** [rOpenGov/geofi#43]
There seems to be a persistent error with pkgdown related to topics missing from index, which prevents new pkgdown site from generating. A recent example: https://github.com/rOpenGov/geofi/runs/4936970331?check_suite_focus=true

```
>-- Building function reference -------------------------------------------------
> 69Error: Error: Topics missing from index: municipality_key_2022
> 70 Backtrace:
> 71 █
> 72 1. └─pkgdown::deploy_to_branch(new_process = FALSE)
> 73 2.   └─pkgdown::build_site_github_pages(pkg, ..., clean = clean)
> 74 3.     └─pkgdown::build_site(...)
> 75 4.       └─pkgdown:::build_site_local(...)
> 76 5.         └─pkgdown::build_reference(...)
> 77 6.           └─pkgdown::build_reference_index(pkg)
> 78 7.             ├─pkgdown::render_page(...)
> 79 8.             │ └─pkgdown:::render_page_html(pkg, name = name, data = data, depth = depth)
> 80 9.             │   └─utils::modifyList(data_template(pkg, depth = depth), data)
> 81 10.             │     └─base::stopifnot(is.list(x), is.list(val))
> 82 11.             └─pkgdown:::data_reference_index(pkg)
> 83 12.               └─pkgdown:::check_missing_topics(rows, pkg)
> 84 11 Removing worktree 11111111111111111111111111111111111111111111111111111111111
> 85 Running git worktree remove /tmp/Rtmpa4n0lp/file33b93695d10f
> 86 Execution halted
> 87 Error: Process completed with exit code 1.
```

Any ideas on how to fix this? There are some related issues in pkgdown repo from 2019 but also from recent days:
https://github.com/r-lib/pkgdown/issues/1132
https://github.com/r-lib/pkgdown/issues/1716
https://github.com/r-lib/pkgdown/issues/1951
https://github.com/r-lib/pkgdown/issues/1958

Changing dataset help files to `@keywords internal` might not be the preferred solution, no?

**Comment by dieghernan** [comment]
The issue is that `municipality_key_2022` is missing from here:


https://github.com/rOpenGov/geofi/blob/9261a41969ab5f70e058541dff0304cda56de0f7/_pkgdown.yml#L19:L31

**Comment by dieghernan** [comment]
There are more topics missed, would have a look

**Comment by pitkant** [comment]
Oh ok, thanks. Most other packages have a very barebones _pkgdown.yml file without custom sub-headers so it didn't occur to me to check it out. It's worth remembering to update it regularly.

**Comment by pitkant** [comment]
There was an additional geofacet that needed to be added. Fixed in db4990b

**Comment by dieghernan** [comment]
A cool trick for avoiding this is to use the roxygen tag @family on a set of common functions. Basically it would link all the functions of the family to the seealso part of each page, and also pkgdown would use it for auto-create the topic index trough has_concept(). See giscoR implementation https://github.com/rOpenGov/giscoR/blob/main/pkgdown/_pkgdown.yml so when I introduce a new function I just have to add the corresponding family tag.

More on https://roxygen2.r-lib.org/articles/rd.html#cross-references and 
https://pkgdown.r-lib.org/reference/build_reference.html#topic-matching
